File: frontend/src/types/prediction.ts
export type Furnishing = "Furnished" | "Semi-Furnished" | "Unfurnished" | "Unknown";
export type Transaction = "New Property" | "Resale" | "Other";
export type AreaSource = "carpet" | "super";

// Mirrors Backend/app/schemas/prediction.py::PredictionRequest exactly —
// field names and types must match, or FastAPI/pydantic will reject the request.
export interface PredictionRequest {
  bhk: number;
  bathroom: number;
  balcony: number;
  area_sqft: number;
  current_floor: number;
  total_floors: number;
  furnishing: Furnishing;
  transaction: Transaction;
  ownership: string;
  facing: string;
  location: string;
  area_source: AreaSource;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    bhk: float = Field(gt=0)
    bathroom: float = Field(ge=0)
    balcony: float = Field(ge=0)
    area_sqft: float = Field(gt=0)
    current_floor: float = Field(ge=0)
    total_floors: float = Field(gt=0)
    furnishing: str
    transaction: str
    ownership: str
    facing: str
    location: str
    area_source: str  # "carpet" or "super"


class PredictionResponse(BaseModel):
    predicted_price: float